### DISCUSSION TITLE
Autocomplete Pandas DataFrame columns using dot-access

### DISCUSSION THREAD

**Discussion by AllanLeanderRostockHansen** [microsoft/pylance-release#2718] [Enhancement]
When working with Pandas' DataFrames, single columns can be accessed using a dot, just like normal object properties.

```python
my_column_dot_access = df.my_column
my_column_index_access = df['my_column']
```
The two lines above return the same data, but the first way is faster to write, and when using a Jupyter Lab notebook, autocompletion are provided for the column names and the methods and properties pertaining to the dot-selected column... it would be awesome if PyLance could do the same thing!

**Comment by savannahostrowski** [comment]
Hey @AllanLeanderRostockHansen,

Thanks for the suggestion! We are currently investigating this enhancement.

**Comment by ma-galan** [comment]
I second this, also when using  "just the brackets" (pd.['name']) or pd.loc['name']
As it happens in Jupyter Lab and Jupyter Notebook  for example, the column names should be accesible when selecting them with the help of Autocompletion.

**Comment by hadim** [comment]
The same issue apply to the omegaconf lib that provides the Omegaconf object which is a dict like object where keys can be accessed by dots similar to pandas dataframe.

**Comment by syagev** [comment]
This would be really awesome. See my related SO question which got some traction and has some relevant materials:
https://stackoverflow.com/questions/55787022/static-typing-schema-of-a-pandas-dataframe

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by marfel** [comment]
Any updates on this enhancement? I would find this very useful, too!

**Comment by surjikal** [comment]
Would be nice for `pyspark` (apache spark) dataframes, too.
